src/nvme/core.c

```diff
@@ -516,11 +577,12 @@ int nvme_pci_init(struct nvme_ctrl *ctrl, const char *bdf)
 	}
 
 	ctrl->pci.classcode = classcode;
-	ctrl->pci.bdf = bdf;
 
 	if (vfio_pci_open(&ctrl->pci, bdf))
 		return -1;
 
+	ctrl->pci.bdf = strdup(bdf);
```

Comment:
Is this a bug fix on its own?

Reply:
Oh, my bad.  It should have not been added.  Will remove it. Nice catch!